crates/starknet/src/starknet/add_invoke_transaction.rs

```diff
@@ -191,30 +205,27 @@ mod tests {
         let (mut starknet, account_address, contract_address, increase_balance_selector, _) =
             setup();
 
+        let nonce = 0;
         let invoke_transaction = test_invoke_transaction_v1(
             account_address,
             contract_address,
             increase_balance_selector,
             Felt::from(10),
-            0,
+            nonce,
         );
 
-        let transaction_hash =
-            starknet.add_invoke_transaction_v1(invoke_transaction.clone()).unwrap();
+        let transaction_hash = starknet.add_invoke_transaction(invoke_transaction.clone()).unwrap();
         let transaction = starknet.transactions.get_by_hash_mut(&transaction_hash).unwrap();
-        assert_eq!(transaction.status, TransactionStatus::AcceptedOnL2);
+        assert_eq!(transaction.finality_status, Some(TransactionFinalityStatus::AcceptedOnL2));
+        assert_eq!(transaction.execution_result.status(), TransactionExecutionStatus::Succeeded);
 
-        let transaction_hash = starknet.add_invoke_transaction_v1(invoke_transaction).unwrap();
+        let transaction_hash = starknet.add_invoke_transaction(invoke_transaction).unwrap();
         let transaction = starknet.transactions.get_by_hash_mut(&transaction_hash).unwrap();
-        assert_eq!(transaction.status, TransactionStatus::Rejected);
-
-        match transaction.execution_error.as_ref().unwrap() {
-            starknet_in_rust::transaction::error::TransactionError::InvalidTransactionNonce(
-                _,
-                _,
-            ) => {}
-            err => panic!("Invalid error type {:?}", err),
-        }
+        assert_eq!(transaction.finality_status, None);
+        assert_eq!(
+            transaction.execution_result.revert_reason(),
+            Some(format!("Invalid transaction nonce. Expected: 1 got {}", nonce).as_str())
```

Comment:
Do we plan to change this with https://github.com/0xSpaceShard/starknet-devnet-rs/issues/150 ?

Reply:
```suggestion
            Some(format!("Invalid transaction nonce. Expected: 1, got {}", nonce).as_str())
```

Reply:
If nonce is part Rejected thing then yes.